src/Gui/Splashscreen.cpp

```diff
@@ -148,76 +113,143 @@ class SplashObserver : public Base::ILogger
 
 /**
  * Constructs a splash screen that will display the pixmap.
  */
-SplashScreen::SplashScreen(  const QPixmap & pixmap , Qt::WindowFlags f )
-    : QSplashScreen(pixmap, f)
+SplashScreen::SplashScreen(const QPixmap& pixmap, Qt::WindowFlags f) : QSplashScreen {pixmap, f}
 {
-    // write the messages to splasher
-    messages = new SplashObserver(this);
+    const std::map<std::string, std::string>& config = App::Application::Config();
+
+    /**
+     * Breaking up code by SRP highlighted conditions that were not taken into account
+     * Even so, it is dubious that the following has to happen here
+     * This is a dependency that could perhaps better be injected
+     */
+
+    std::string colStr = fromConfig(config, "SplashTextColor");
+    if (!colStr.empty()) {
+        auto color = colorFromString(colStr);
+        if (color.isValid()) {
+            textColor = color;
+        }
+    }
+
+    std::string alignStr = fromConfig(config, "SplashAlignment");
+    if (!alignStr.empty()) {
+        alignment = alignStrToInt(alignStr);
+    }
+
+    splashObserver = new SplashObserver(*this);
 }
 
 /** Destruction. */
 SplashScreen::~SplashScreen()
 {
-    delete messages;
+    delete splashObserver;
+}
+
+void SplashScreen::showMessage(const QString& message)
+{
+    QSplashScreen::showMessage(message, alignment, textColor);
+}
+
+std::string SplashScreen::fromConfig(const std::map<std::string, std::string>& config,
+                                     const std::string value)
+{
+    auto al = config.find(value);
+    return al == config.end() ? "" : al->second.c_str();
+}
+
+QColor SplashScreen::colorFromString(const std::string str)
+{
+    QColor color {};
+    color.setNamedColor(QString::fromStdString(str));
+    return color;
+}
+
+int SplashScreen::alignStrToInt(const std::string str)
+{
+    // c++20 has .ends_with and .starts_with !!
+    QString qstr {QString::fromStdString(str)};
+    int align {};
+    if (qstr.startsWith(QLatin1String("VCenter"))) {
+        align = Qt::AlignVCenter;
+    }
+    else if (qstr.startsWith(QLatin1String("Top"))) {
+        align = Qt::AlignTop;
+    }
+    else {
+        align = Qt::AlignBottom;
+    }
+
+    if (qstr.endsWith(QLatin1String("HCenter"))) {
+        align += Qt::AlignHCenter;
+    }
+    else if (qstr.endsWith(QLatin1String("Right"))) {
+        align += Qt::AlignRight;
+    }
+    else {
+        align += Qt::AlignLeft;
+    }
+    return align;
 }
 
 /**
  * Draws the contents of the splash screen using painter \a painter. The default
  * implementation draws the message passed by message().
  */
-void SplashScreen::drawContents ( QPainter * painter )
+void SplashScreen::drawContents(QPainter* painter)
 {
     QSplashScreen::drawContents(painter);
 }
 
 // ------------------------------------------------------------------------------
 
-AboutDialogFactory* AboutDialogFactory::factory = nullptr;
-
-AboutDialogFactory::~AboutDialogFactory()

```

Comment:
What happened to the definition of the destructor? I cannot find it anywhere. Since AboutDialogFactory has a virtual method and doesn't inherit from another class it must have a virtual destructor.